# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1347677820 14400
#      Fri Sep 14 22:57:00 2012 -0400
# Node ID 1ddf5772fb1a94b446ae177b375a8dab8908bfbe
# Parent  ad2c3902b826b62818e55046fcbac605e7c8d57c
fix broken libgui build

* libgui/Makefile.am (liboctgui_la_SOURCES): New variable.
* libgui/src/module.mk (src_libgui_src_la_SOURCES): Rename from
liboctgui_la_SOURCES.
(nodist_src_libgui_src_la_SOURCES): Rename from
nodist_liboctgui_la_SOURCES.
(src_libgui_src_la_CPPFLAGS): Rename from liboctgui_la_CPPFLAGS.
(src_libgui_src_la_CFLAGS): Rename from liboctgui_la_CFLAGS.
(src_libgui_src_la_CXXFLAGS): Rename from liboctgui_la_CXXFLAGS.

diff --git a/libgui/Makefile.am b/libgui/Makefile.am
--- a/libgui/Makefile.am
+++ b/libgui/Makefile.am
@@ -42,16 +42,20 @@ CLEANFILES =
 
 noinst_HEADERS =
 
 noinst_LTLIBRARIES =
 
 include src/module.mk
 include qterminal-module.mk
 
+## libocgui merely collects of compiled convenience libraries.
+## It has no source code itself.
+liboctgui_la_SOURCES = 
+
 include link-deps.mk
 
 # Dummy C++ source to cause C++ linking.
 nodist_EXTRA_liboctgui_la_SOURCES = dummy.cc
 
 liboctgui_la_LIBADD = \
   qterminal/libqterminal.la \
   src/libgui-src.la \
diff --git a/libgui/src/module.mk b/libgui/src/module.mk
--- a/libgui/src/module.mk
+++ b/libgui/src/module.mk
@@ -27,17 +27,17 @@ octave_gui_MOC = \
 octave_gui_UI = \
   src/settings-dialog.ui \
   src/welcome-wizard.ui
 
 octave_gui_UI_H = $(patsubst src/%.ui, src/ui-%.h, $(octave_gui_UI))
 
 octave_gui_RC = src/qrc-resource.cc
 
-liboctgui_la_SOURCES = \
+src_libgui_src_la_SOURCES = \
   src/documentation-dockwidget.cc \
   src/files-dockwidget.cc \
   src/history-dockwidget.cc \
   src/m-editor/file-editor-tab.cc \
   src/m-editor/file-editor.cc \
   src/m-editor/find-dialog.cc \
   src/m-editor/lexer-octave-gui.cc \
   src/main-window.cc \
@@ -51,17 +51,17 @@ liboctgui_la_SOURCES = \
   src/resource-manager.cc \
   src/settings-dialog.cc \
   src/symbol-information.cc \
   src/terminal-dockwidget.cc \
   src/welcome-wizard.cc \
   src/workspace-model.cc \
   src/workspace-view.cc
 
-nodist_liboctgui_la_SOURCES = $(octave_gui_MOC) $(octave_gui_RC)
+nodist_src_libgui_src_la_SOURCES = $(octave_gui_MOC) $(octave_gui_RC)
 
 noinst_HEADERS += \
   src/documentation-dockwidget.h \
   src/files-dockwidget.h \
   src/history-dockwidget.h \
   src/m-editor/file-editor-interface.h \
   src/m-editor/file-editor-tab.h \
   src/m-editor/file-editor.h \
@@ -126,17 +126,17 @@ octave_gui_ICONS = \
   src/icons/star.png \
   src/icons/stop.png \
   src/icons/terminal.png \
   src/icons/undo.png \
   src/icons/up.png \
   src/icons/zoom-in.png \
   src/icons/zoom-out.png
 
-liboctgui_la_CPPFLAGS = \
+src_libgui_src_la_CPPFLAGS = \
   $(AM_CPPFLAGS) \
   @OCTGUI_DLL_DEFS@ \
   -I$(QT_INCDIR) \
   -I$(QT_INCDIR)/QtCore \
   -I$(QT_INCDIR)/QtGui \
   -I$(QT_INCDIR)/QtNetwork \
   -I$(srcdir)/qterminal/libqterminal \
   -Isrc -I$(srcdir)/src \
@@ -150,16 +150,13 @@ liboctgui_la_CPPFLAGS = \
   -I$(top_srcdir)/liboctave/system \
   -I$(top_srcdir)/liboctave/util \
   -I$(top_srcdir)/libinterp \
   -I$(top_builddir)/libinterp/parse-tree -I$(top_srcdir)/libinterp/parse-tree \
   -I$(top_builddir)/libinterp/interp-core -I$(top_srcdir)/libinterp/interp-core \
   -I$(top_builddir)/libinterp/interpfcn -I$(top_srcdir)/libinterp/interpfcn \
   -I$(top_srcdir)/libinterp/octave-value
 
-liboctgui_la_CFLAGS = $(AM_CFLAGS) $(WARN_CFLAGS)
+src_libgui_src_la_CFLAGS = $(AM_CFLAGS) $(WARN_CFLAGS)
 
-liboctgui_la_CXXFLAGS = $(AM_CXXFLAGS) $(WARN_CXXFLAGS)
+src_libgui_src_la_CXXFLAGS = $(AM_CXXFLAGS) $(WARN_CXXFLAGS)
 
 noinst_LTLIBRARIES += src/libgui-src.la
-
-src_libgui_src_la_SOURCES = $(GUI_SRC_SRC)
-src_libgui_src_la_CPPFLAGS = $(liboctgui_la_CPPFLAGS)
